docs/concepts/caching.rst

```diff
@@ -0,0 +1,489 @@
+
+Using the Hamilton ``@tag`` decorator, you can cache a node's result to a specific file format (``JSON``, ``CSV``, ``Parquet``, etc.) and circumvent the limitations of ``pickle`` files.
+
+
+.. code-block:: python
+
+    # my_dataflow.py
+    import pandas as pd
+    from hamilton.function_modifiers import tag
+
+    def raw_data(path: str) -> pd.DataFrame:
+        return pd.read_csv(path)
+
+    @tag(cache="parquet")
+    def clean_dataset(raw_data: pd.DataFrame) -> pd.DataFrame:
+        raw_data = raw_data.fillna(0)
+        return raw_data
+
+    @tag(cache="json")
+    def statistics(clean_dataset: pd.DataFrame) -> dict:
+        return ...
+
+
+.. note::
+
+    This aims to match and replace the features of ``hamilton.experimental.h_cache.CachingGraphAdater`` but the API is preliminary.
+
+
+Resume from checkpoint
+-----------------------
+
+The caching feature stores results after each node execution. If execution fails half-way, the ``Driver`` can use the cache metadata and results to resume the run. You can think of it as automatically passing ``.execute(overrides={...})`` from disk.
+
+To resume, you can specify ``.with_cache(resume_from=...)`` which accepts a specific ``run_id`` or the string ``"latest"`` which will automatically get the results from the latest execution, whether it succeeded or failed. The ``run_id`` is automatically generated internally and can be viewed via the cache logger.
+
+.. code-block:: python
+
+    from hamilton import driver
+    import my_dataflow
+
+    dr = (
+        driver.Builder()
+        .with_modules(my_dataflow)
+        .with_cache(resume_from="latest")
+        .build()
+    )
+
+    dr.execute(...)
+
+.. note::
+
+    Using ``resume_from="latest"`` gets the latest run at the moment of building the ``Driver``. Consequently, calling ``Driver.execute()`` multiple times afterwards will always retrieve the same results from that specific point in time.
+
+
+Control caching behavior
+-------------------------
+
+Hamilton's default caching behavior is ideal to get started and typically what you want during iterative development. However, you might want more control over the cache behavior in specific scenario or when moving to production. The caching behavior can be specified at the node-level via either:
+
+- **Dataflow definition** by adding to the function modifiers ``@tag(caching_behavior="ignore")``
+- **Driver definition** by passing a list of node name to ``.with_cache(ignore=["foo", "bar"])``
+
+The "Driver definition" approach will always override the behavior specified using ``@tag``. Passing empty lists to ``ignore=[]`` effectively disables all ``@tag``. Querying node names directly from the ``Driver`` is an effective way to select multiple nodes at once:
```

Comment:
we probably want `None` here versus empty list.

Reply:
or instead be explicit:
```python
.with_cache(ignore="*")
```

Reply:
I believe what I meant here is the opposite of what you describe. The section refers to a dataflow like this

```python
@tag(cache_behavior="ignore")
def  foo() -> int: ...

def bar(foo: int) -> int: ...
```

What if the user wants to "disable" `cache="ignore"` from the `Driver` code? To me, having
```python
dr = driver.Builder().with_modules(...).with_cache(ignore=[ ])
```
Means "disregard all `cache_behavior="ignore"` statements". Since there's more than 2 behaviors, "disregard all `ignore`" isn't equivalent to "set all to `default`". 

```python
.with_cache(ignore="*")
#not equal to 
.with_cache(default="*")
```